### DISCUSSION TITLE
Expensive time to do finetune (HELP)

### DISCUSSION THREAD

**Discussion by fliperaa** [erew123/alltalk_tts#401] [Q&A] [answered]
sorry anyone can help me i run finetune in my PC with an 9minute good quality audio, but in the first step "generating the dataset" it's locked in 75%, was runned 36986.8s and no complete. 

my pc is an 
r5 3600
rtx2060
16gb ram

i understand my pc maybe is limited, but this is a very long run time no ? 

i used whisper large model

when i start finetune, receive this msg to att gradio: 
IMPORTANT: You are using gradio version 4.32.2, however version 4.44.1 is available, please upgrade.

Really need upgrade ?

**Comment by erew123** [comment]
Hi @fliperaa The gradio version is a nothing, so dont worry about that.

As for the getting stuck at 75% it could be an out of memory issue OR a corrupt file.

I cant speak as to your dataset/audio files, however what I can offer is a file to test with.

https://filetransfer.io/data-package/woURvHoB

If you want to put this and ONLY this file in your `put-voice-samples-in-here` folder and test generating your dataset with a new project name:

![image](https://github.com/user-attachments/assets/ee36c045-e97a-477c-b8be-7b22a65131d1)

I cant say exactly how long this would take on your hardware, but on a Nvidia 4070, its about a 2-5 minute task for it to break that file up, so I would image 4-7 minutes at a guess (on your hardware).

At least if you try this out, you can identify is it a hardware issue (memory etc) or is it (probably) a audio samples/dataset issue.

Thanks

**Reply by fliperaa** [reply]
hey @erew123, I tested data generation with your file and it worked successfully, thank you very much!

As for my audio, I edited it in Audacity and exported it in WAV, any guesses as to what could have gone wrong with the file?

What is the best way to obtain a test file in Portuguese to use finetune?

**Reply by fliperaa** [reply]
@erew123 I had a new problem, I think my files are really corrupt, in the training stage, I got this error: 

How i solve that ?

`[FINETUNE] The text length exceeds the character limit of 250 for language 'en', this might cause truncated audio.
[FINETUNE] The text length exceeds the character limit of 250 for language 'en', this might cause truncated audio.
formers\pytorch_utils.py", line 111, in forward
    x = torch.addmm(self.bias, x.view(-1, x.size(-1)), self.weight)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 14.00 MiB. GPU 0 has a total capacity of 6.00 GiB of which 0 bytes is free. Of the allocated memory 8.09 GiB is allocated by PyTorch, and 162.60 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
ERROR:    Traceback (most recent call last):
  File "C:\Users\FLIP\alltalk_tts\trainer_alltalk\trainer.py", line 1875, in fit
    self._fit()
.......................etc..... 
  File "C:\Users\FLIP\alltalk_tts\alltalk_environment\env\Lib\site-packages\uvicorn\protocols\http\h11_impl.py", line 534, in receive
    await self.message_event.wait()
  File "C:\Users\FLIP\alltalk_tts\alltalk_environment\env\Lib\asyncio\locks.py", line 213, in wait
    await fut
asyncio.exceptions.CancelledError`

**Reply by erew123** [reply]
@fliperaa No idea with your audio files. I would run them through some software to convert them again so that they are ok. Maybe they are corrupted, maybe they are just too small. They should preferably be over 6 seconds in length. And I have no idea where to find Portuguese audio, Maybe have a hunt of the Coqui forums https://github.com/coqui-ai/TTS/discussions

Re your error, please see this: https://github.com/erew123/alltalk_tts/wiki/Error-Messages-List#general-issues

**Reply by fliperaa** [reply]
@erew123 I came to the conclusion that perhaps the problem is in whisper, when I try to generate the dataset with my audio using distil-large-v3(en only) it works, even though my audio is in pt, and when I try to play the file you sent me using the normal whisper does not (en only) get stuck and does not generate! 

Later I will try to reinstall alltalks completely, to see if it solves it!

In any case, I would appreciate any suggestions for a solution, it is very difficult to find a model to improve your PT-BR voice, I really want to do that! 

Thank you very much for your attention! 

Note: if you want to try running my pt-br file to test, I'll attach it here!
https://filetransfer.io/data-package/2l68mlYO#link

**Reply by erew123** [reply]
@fliperaa There is an entirely new, from the ground up, rebuilt version of finetune that you can now use, You will find it better with foreign language dataset building (as long as you use the right whisper model). You may have to try different models and vad on or off. https://github.com/erew123/alltalk_tts/tree/alltalkbeta?tab=readme-ov-file#-updating

**Reply by erew123** [reply]
@fliperaa I have created your dataset for you with the new finetune. If you update your finetune, you can import/use this dataset to finetune. https://filetransfer.io/data-package/s58LO2o0#link

![image](https://github.com/user-attachments/assets/07b1aecb-30fb-4071-849f-57ac85e68c36)

**Reply by fliperaa** [reply]
@erew123 I formatted my computer completely and reinstalled the updated alltalks, now I can create my dataset perfectly in finetune, but when I get to training I get this message here: what do I do?
![image](https://github.com/user-attachments/assets/b7ada609-c259-43a4-b7ad-cf308138b5b1)

**Reply by erew123** [reply]
@fliperaa when did you last update alltalk? I posted an update about 5 hours ago to fine-tune. There should be no bugs, but lmk. If not, please git pull the most recent update.

**Reply by fliperaa** [reply]
Oh nice,I hadn't seen the new update, I'll update it here and test it, thanks!

**Reply by erew123** [reply, marked as answer]
@fliperaa can confirm current build is working fine w/training and logging metrics too:

![image](https://github.com/user-attachments/assets/4995be9f-5ef7-4e8b-90a1-936f3976c20a)

As mentioned this has been a huge re-write part of which is documentation, so you should find most answers to any questions within the built in help or in the WIKI sections https://github.com/erew123/alltalk_tts/wiki#-xtts-finetuning-guides should you need more detail.

Thanks

**Reply by fliperaa** [reply]
@erew123 thanks bro, now it's working 100% correct!
